Mathlib/Data/List/Infix.lean

```diff
@@ -507,4 +526,14 @@ theorem mem_of_mem_suffix (hx : a ∈ l₁) (hl : l₁ <:+ l₂) : a ∈ l₂ :=
   hl.subset hx
 #align list.mem_of_mem_suffix List.mem_of_mem_suffix
 
+theorem ne_nil_prefix {x y : List α} (hx : x ≠ []) (h : x <+: y) : y ≠ [] := by
+  rintro rfl; apply hx; exact List.prefix_nil.mp h
```

Comment:
Often, the first version of a proof can be "golfed", i.e. written in a shorter or more concise way. This is such a case: the last lines can be combined into one: `exact hx (List.prefix_nil.mp h)`